src/frontend/src/lib/services/rest.services.ts

```diff
@@ -10,35 +8,29 @@ export interface RestRequestParams<Response, Error = unknown, Success = void> {
  * A utility function to make a REST request with possible retries.
  *
  * @param request - The request function to call.
- * @param onSuccess - The function to call when the request is successful.
- * @param onError - The optional function to call when the request fails and the max retries are reached.
  * @param onRetry - The optional function to call when the request fails and a retry is needed.
  * @param maxRetries - The maximum number of retries to attempt.
- * @returns The result of the onSuccess/onError function or undefined if the max retries are reached.
+ * @returns The result of the request or it throws an error if the max retries are reached.
  */
-export const restRequest = async <Response, Error, Success>({
+export const restRequest = async <Response, Error>({
 	request,
-	onSuccess,
-	onError,
 	onRetry,
 	maxRetries = 3
-}: RestRequestParams<Response, Error, Success>): Promise<Success | undefined> => {
-	let retryCount = 0;
-
-	while (retryCount <= maxRetries) {
+}: RestRequestParams<Response, Error>): Promise<Response | undefined> => {
+	const executeRequest = async (retryCount: number): Promise<Response> => {
 		try {
-			const response = await request();
-			return onSuccess(response);
+			return await request();
 		} catch (error: unknown) {
-			retryCount++;
-
-			if (retryCount > maxRetries) {
+			if (retryCount >= maxRetries) {
 				console.error(`Max retries reached. Error:`, error);
-				return onError?.(error as Error);
+				throw error;
 			}
 
 			await onRetry?.({ error: error as Error, retryCount });
-			console.warn(`Request attempt ${retryCount} failed. Retrying...`);
+			console.warn(`Request attempt ${retryCount + 1} failed. Retrying...`);
+			return executeRequest(retryCount + 1);
```

Comment:
isn't the keyword `await` missing here?

Reply:
very true! strange that it was not flagged...